src/main/resources/static/userRequestFunctions.js

```diff
@@ -0,0 +1,70 @@
+function postUser() {
+
+    var name = document.getElementById("newName").value;
+    var postRequest = new XMLHttpRequest();
+
+    postRequest.open("POST", 'users/' + name, false);
+
+    postRequest.send();
```

Comment:
&#x1F34E; **Issue**
It looks like a synchronous request. Browser executes your JavaScript in one thread. That means your application will hang upon such a request. It's not very user-friendly. Usually, every heavy-weight operation, in the environments like this one, is done on another thread (or, as in case of JS through a callback) - i.e. asynchronously.